x/incentives/keeper/grpc_query.go

```diff
@@ -193,6 +193,63 @@ func (k Keeper) GasMeter(
 	return &types.QueryGasMeterResponse{GasMeter: gm}, nil
 }
 
+// AllocationMeters return registered allocation meters
+func (k Keeper) AllocationMeters(
+	c context.Context,
+	req *types.QueryAllocationMetersRequest,
+) (*types.QueryAllocationMetersResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+
+	var allocationMeters []sdk.DecCoin
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixAllocationMeter)
+
+	pageRes, err := query.Paginate(
+		store,
+		req.Pagination,
+		func(denom, value []byte) error {
+			allocationMeter, found := k.GetAllocationMeter(ctx, string(denom))
+			if found {
+				allocationMeters = append(allocationMeters, allocationMeter)
+			}
```

Comment:
you don't need to get the allocation, just `Unmarshal` the value

Reply:
@fedekunze trying to unmarshal the `sdk.DecCoin` like this: 

```
			var allocationMeter sdk.DecCoin
			if err := allocationMeter.Unmarshal(value); err != nil {
				return err
			}
```

but getting nil as allocationMeter

Reply:
should be 

```golang
var amount sdk.Dec
if err := amount.Unmarshal(amount); err != nil {
    return err
}
```